examples/app/uploadservice-s3/src/main/java/com/example/uploadservice/s3/S3UploadTask.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.uploadservice.s3
+
+import android.util.Log
+import com.amazonaws.auth.CognitoCachingCredentialsProvider
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferListener
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferNetworkLossHandler
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferState
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferUtility
+import com.amazonaws.regions.Region
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.s3.AmazonS3Client
+import com.amazonaws.services.s3.model.CannedAccessControlList
+import net.gotev.uploadservice.UploadTask
+import net.gotev.uploadservice.network.HttpStack
+import net.gotev.uploadservice.network.ServerResponse
+import java.io.File
+
+class S3UploadTask : UploadTask() {
+
+    private val s3params by lazy {
+        S3UploadTaskParameters.createFromPersistableData(params.additionalParameters)
+    }
+
+    override fun upload(httpStack: HttpStack) {
+        val s3params = s3params
+        val credentialsProvider = CognitoCachingCredentialsProvider(context, s3params.identityPoolId, Regions.fromName(s3params.region))
+        val s3 = AmazonS3Client(credentialsProvider, Region.getRegion(Regions.fromName(s3params.region)))
+        s3.getUrl(s3params.bucket_name, s3params.serverSubpath)
+        val transferUtility = TransferUtility.builder().s3Client(s3).context(context).build()
+        val file = File(s3params.uploadFilepath)
+        require(file.exists()) { "Error! Please choose a valid file for upload" }
+        TransferNetworkLossHandler.getInstance(context)
+        val observer = transferUtility.upload(
+                s3params.bucket_name,
+                s3params.serverSubpath,
+                file,
+                CannedAccessControlList.Private
```

Comment:
This aws stuff could/should be moved into its own class similar to ftpclientwrapper.